### PR TITLE
[AzDatalake] Directory Client Implementation

### PR DESCRIPTION
**Pull request by tasherif-msft**
More tests in separate PR

### CODE REVIEW

File: sdk/storage/azdatalake/directory/client.go [hunk 1]
```diff
@@ -174,65 +185,189 @@ func (d *Client) BlobURL() string {
 
 //TODO: create method to get file client - this will require block blob to have a method to get another block blob
 
-// Create creates a new directory (dfs1).
+// Create creates a new file (dfs1).
```
Comment (gapra-msft) [inline sdk/storage/azdatalake/directory/client.go:188]:
```suggestion
// Create creates a new directory (dfs1).
```

File: sdk/storage/azdatalake/directory/client.go [hunk 2]
```diff
@@ -174,65 +185,189 @@ func (d *Client) BlobURL() string {
...
-	//TODO: pass recursive = true
-	return DeleteResponse{}, nil
+	lac, mac, deleteOpts := path.FormatDeleteOptions(options, true)
+	resp, err := d.generatedDirClientWithDFS().Delete(ctx, deleteOpts, lac, mac)
+	err = exported.ConvertToDFSError(err)
+	return resp, err
 }
 
-// GetProperties returns the properties of the directory (blob3). #TODO: we may just move this method to path client
+// GetProperties gets the properties of a file (blob3)
```
Comment (gapra-msft) [inline sdk/storage/azdatalake/directory/client.go:204]:
```suggestion
// GetProperties gets the properties of a directory (blob3)
```

File: sdk/storage/azdatalake/directory/client.go [hunk 3]
```diff
@@ -174,65 +185,189 @@ func (d *Client) BlobURL() string {
...
+	lastIndex := strings.LastIndex(endpoint, separator)
+	// Split the string based on the last occurrence of the separator
+	firstPart := endpoint[:lastIndex] // From the beginning of the string to the last occurrence of the separator
+	newPathURL, newBlobURL := shared.GetURLs(runtime.JoinPaths(firstPart, newName))
+	parsedNewURL, _ := url.Parse(d.DFSURL())
+	return parsedNewURL.Path, newPathURL, newBlobURL
 }
 
-// Rename renames the directory (dfs1).
+// Rename renames a file (dfs1)
```
Comment (gapra-msft) [inline sdk/storage/azdatalake/directory/client.go:227]:
```suggestion
// Rename renames a directory (dfs1)
```

File: sdk/storage/azdatalake/directory/client.go [hunk 4]
```diff
@@ -174,65 +185,189 @@ func (d *Client) BlobURL() string {
...
+	opts := path.FormatGetPropertiesOptions(options)
+	var respFromCtx *http.Response
+	ctxWithResp := runtime.WithCaptureResponse(ctx, &respFromCtx)
+	resp, err := d.blobClient().GetProperties(ctxWithResp, opts)
+	newResp := path.FormatGetPropertiesResponse(&resp, respFromCtx)
+	err = exported.ConvertToDFSError(err)
+	return newResp, err
+}
+
+func (d *Client) renamePathInURL(newName string) (string, string, string) {
```
Comment (gapra-msft) [inline sdk/storage/azdatalake/directory/client.go:215]: can we make sure to add individual UT for this method?

**Review by gapra-msft — APPROVED** [review]
lgtm just had a few comments

File: sdk/storage/azdatalake/directory/client.go (hunk 4 above)
Comment (tasherif-msft) [reply sdk/storage/azdatalake/directory/client.go:215]: absolutely will put as TODO

**Pull request merged**